content/doc/book/pipeline/scaling-pipeline.adoc

```diff
@@ -0,0 +1,88 @@
+---
+layout: section
+title: Scaling Pipelines
+---
+:description:
+:author: Sam Van Oort
+:sectanchors:
+:toc:
+:hide-uri-scheme:
+
+
+= Using Speed/Durability Settings To Reduce Disk I/O Needs
+
+One of the main bottlenecks in Pipeline is that it writes transient data to disk *FREQUENTLY* in order to be able to resume Pipelines when Jenkins restarts or if something goes wrong and it crashes.  This durability is useful for many users but its performance cost can be a problem. 
+
+Pipeline now includes features to let users improve performance by reducing how much data is written to disk and how often it is written -- at a small cost to durability.  In some special cases, users may not be able to resume or visualize running Pipelines if Jenkins shuts down suddenly without getting a chance to write data. 
+
+Because these settings include a trade-off they are initially opt-in. Users need to explicity set a _Speed/Durability Setting_ for Pipelines.  Initially they will default to the least aggressive setting.  Note that the same plugin updates including these new settings also include additional I/O optimizations. 
+
+== How Do I Set Speed/Durability Settings?
+There are 3 ways to configure the durability setting:
+
+. *Globally*, you can choose a global default durability setting under "Manage Jenkins" > "Configure System", labelled "Pipeline Speed/Durability Settings".  You can override these with the more specific settings below.
+
+. *Per pipeline job:* at the top of the job configuration, labelled "Custom Pipeline Speed/Durability Level" - this overrides the global setting.  Or, use a "properties" step - the setting will apply to the NEXT run after the step is executed (same result).
+
+. *Per-branch for a multibranch project:* configure a custom Branch Property Strategy (under the SCM) and add a property for Custom Pipeline Speed/Durability Level.  This overrides the global setting. You can also use a "properties" step to override the setting, but remember that you may have to run the step again to undo this.
+
+Durability settings will take effect with the next applicable Pipeline run, not immediately.  The setting will be displayed in the log. 
+
+== Will Higher-Performance Durability Settings Help Me?
+* Yes, if you are running complex Pipelines or Pipelines with many steps (more than several hundred).
+* Yes, if your Pipeline stores large files or complex data to variables in the script and then runs steps.  This sounds oddly specific but happens more than you'd expect.
```

Comment:
Perhaps provide an example, such as parsing a big JSON file using https://jenkins.io/doc/pipeline/steps/pipeline-utility-steps/#code-readjson-code-read-json-from-files-in-the-workspace ? Or just readFile?